### PR TITLE
add spurious protein info

### PR DESCRIPTION
**Pull request by typhainepl**

### CODE REVIEW

**Review by matthiasblum — CHANGES_REQUESTED** [review]
Well done!

There are a couple of issues, but they should be easy to address.

File: pronto/static/js/ui/proteins.js [hunk 1]
```diff
@@ -24,29 +24,30 @@ export function genProtHeader(protein) {
             Organism: <em>${protein.organism.name}</em>
             &mdash;
             Length: ${protein.length} AA
-            </div>`;
+            ${protein.is_spurious ? '&mdash; <span class="ui spurious"><i class="warning icon"></i >Spurious protein</span>' : ''}
+            </div > `;
```
[the remark this answers is not in the document]
Comment (typhainepl) [reply pronto/static/js/ui/proteins.js:28]: For the first part, I thought about it, but I also wanted the info to be visible when looking at the overlapping page, and I thought having both on the protein page will be duplicating information.

Second part, the text isn't 100% red atm, I had chosen another colour when playing around with the message box. I will change it to use `ui red text`. I didn't know which css to use, I'll be careful in the future

File: pronto/api/protein.py [hunk 2]
```diff
@@ -24,18 +24,23 @@ def get_protein(protein_acc):
...
-               p.is_reviewed, t.id, t.name, pn.text
+               p.is_reviewed, t.id, t.name, pn.text, m.protein_acc
         FROM protein p
         INNER JOIN taxon t ON p.taxon_id = t.id
         INNER JOIN protein2name p2n ON p.accession = p2n.protein_acc
         INNER JOIN protein_name pn ON p2n.name_id = pn.name_id
+        LEFT JOIN 
+            (SELECT protein_acc 
+            FROM match 
+            WHERE database_id=16) m ON m.protein_acc=p.accession
```
[the remark this answers is not in the document]
Comment (typhainepl) [reply pronto/api/protein.py]: done

File: pronto/static/js/ui/proteins.js [hunk 3]
```diff
@@ -24,29 +24,30 @@ export function genProtHeader(protein) {
...
-                }
+            html += '<g>';
+            if (i) {
+                // Discontinuous domain: draw arc
+                const px = Math.round(fragments[i - 1].end * width / proteinLength) + paddingLeft;
+                html += `< path d = "M${px} 15 Q ${(px + x) / 2} 0 ${x} 15" fill = "none" stroke = "${signature.color}" /> `
+            }
 
-                html += `<rect x="${x}" y="15" width="${w}" height="10" rx="1" ry="1" style="fill: ${signature.color};" />
+            html += `< rect x = "${x}" y = "15" width = "${w}" height = "10" rx = "1" ry = "1" style = "fill: ${signature.color};" />
```
[the remark this answers is not in the document]
Comment (typhainepl) [reply pronto/static/js/ui/proteins.js]: resolved

File: pronto/static/js/ui/proteins.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthiasblum) [reply pronto/static/js/ui/proteins.js]: I don't think there would be duplication of information as `protein.js` is for the protein page, and `signatures.js` is for the overlapping proteins page. But it is fine as it is, the information is there.

Thank you for reverting the `semantic.css` file. It's true that without a proper developer documentation, you couldn't know which file to edit. Sorry about that. :sweat_smile:

File: pronto/api/protein.py [hunk 4]
```diff
@@ -26,18 +26,26 @@ def get_protein(protein_acc):
...
+        """
+            SELECT COUNT(*) 
+            FROM match 
+            WHERE protein_acc = UPPER(%s)
+            AND database_id = (SELECT id FROM database WHERE name = 'antifam')
+        """,
+        (protein_acc,),
+    )
+    row = cur.fetchone()
+    is_spurious = True if row[0] != 0 else False
```
Comment (matthiasblum) [inline pronto/api/protein.py:39]: As `row[0] != 0` returns a boolean, you can directly do:
```
is_spurious = row[0] != 0
```

**Review by matthiasblum — CHANGES_REQUESTED** [review]
Almost there! Only one tiny change to make, and it's only because I am being picky. ;)

**Review by matthiasblum — APPROVED** [review]
Good to go. Thank you!

**Pull request merged**